### ISSUE TITLE
Incorrect indentation on line starting with &nbsp;

### ISSUE THREAD

**Issue by reidab** [ember-template-lint/ember-template-lint#120]
Linting a template with a segment like the following incorrectly flags the indentation:

``` handlebars
{{#if foo}}
  &nbsp;bar
{{/if}}
```

with

```
block-indentation: Incorrect indentation for `   bar
` beginning at L2:C0. Expected `   bar
` to be at an indentation of 2 but was found at 3. (thedyrt-ui/templates/components/at-trip):
`{{#if foo}}
  &nbsp;bar
{{/if}}`
```

**Comment by duggiefresh** [comment]
What would be the expectation here? Thanks!

**Comment by reidab** [comment]
It's looking like ember-template-lint is treating the `&nbsp;` string as a space for the purpose of calculating source code indentation, instead of treating the `&` as the first character of that line's source.

I'd expect both of the following to pass linting, but currently only the second one does:

``` handlebars
{{#if foo}}
  &nbsp;bar
{{/if}}
```
``` handlebars
{{#if foo}}
  &rarr;bar
{{/if}}
```

**Comment by rwjblue** [comment]
Agreed.  I am unsure why `&nbsp;` is being treated differently than `&rarr;` off the top of my head.

**Comment by rwjblue** [comment]
@reidab - Can you double check if this is still an issue, and open a failing test case for this (basically like [this one](https://github.com/rwjblue/ember-template-lint/blob/master/test/unit/rules/lint-block-indentation-test.js#L89-L92)) if it is still an issue?

**Comment by reidab** [comment]
Confirmed that it's still happening on the latest version and submitted failing tests in #184.

**Comment by rwjblue** [comment]
We can fix it now! We need to update our usage of @glimmer/compiler to use the new `mode: 'codemod'` and it will opt-out of converting HTML entities to their underlying thing. 

See https://github.com/glimmerjs/glimmer-vm/pull/938 for background on that new parsing mode.
